### PR TITLE
fix(button): #DRIV-58 cancel sharing is now working

### PR DESCRIPTION
**Pull request by aymericdevriesere**
## Describe your changes
Cancel button is now working.
## Checklist tests
-[x] Cancel share 
## Issue ticket number and link
[ DRIV-58 ]
https://entsupport.gdapublic.fr/browse/DRIV-58
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/sniplets/content/workspace-nextcloud-toolbar-share.sniplet.ts [hunk 1]
```diff
@@ -74,7 +74,10 @@ export class ToolbarShareSnipletViewModel implements IViewModel {
 
     async onCancelShareElements(): Promise<void> {
         if (this.sharedElement.length) {
-            await service.deleteAll(this.sharedElement);
+            service.deleteAll(this.sharedElement)
```
Comment (JinDouang) [inline src/main/resources/public/ts/sniplets/content/workspace-nextcloud-toolbar-share.sniplet.ts:77]: qu'est-ce qui a changé ?
C'était quoi le pb ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/sniplets/content/workspace-nextcloud-toolbar-share.sniplet.ts (hunk 1 above)
Comment (aymericdevriesere) [reply src/main/resources/public/ts/sniplets/content/workspace-nextcloud-toolbar-share.sniplet.ts:77]: Qaund ça fail avec le await on ne passait pas à la suite.

Comment (JinDouang) [reply src/main/resources/public/ts/sniplets/content/workspace-nextcloud-toolbar-share.sniplet.ts:77]: try catch dans ce cas

```suggestion
  try {
     await service.deleteAll(this.sharedElement);
  } catch (e) {
  console.error((err));
  }
```
On sait quelles sont les types d'erreurs ?

Comment (aymericdevriesere) [reply src/main/resources/public/ts/sniplets/content/workspace-nextcloud-toolbar-share.sniplet.ts:77]: ok je fais ça

**Review by JinDouang — APPROVED** [review]

File: src/main/resources/public/ts/sniplets/content/workspace-nextcloud-toolbar-share.sniplet.ts [hunk 2]
```diff
@@ -70,12 +70,18 @@ export class ToolbarShareSnipletViewModel implements IViewModel {
...
     async onCancelShareElements(): Promise<void> {
         if (this.sharedElement.length) {
-            await service.deleteAll(this.sharedElement);
+            try {
+                await service.deleteAll(this.sharedElement);
+            } catch (e) {
+                console.error((e));
+            }
         }
+        this.sharedElement = [];
```
Comment (JinDouang) [inline src/main/resources/public/ts/sniplets/content/workspace-nextcloud-toolbar-share.sniplet.ts:84]: Pourquoi pas dans le if ?
Si tu es pas dans le if, il est forcément vide et tu le réassigne à une tableau vide

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**